### DISCUSSION TITLE
Building with Vite

### DISCUSSION THREAD

**Discussion by Ehesp** [suren-atoyan/monaco-react#179] [General]
This isn't an issue since I don't believe it to ben an issue with this library, however using this package with Vite only works during development.

The Monaco package imports CSS directly within the JS files, which according to [this issue](https://github.com/microsoft/monaco-editor/issues/886) is not valid ESM module standard.

During build (`vite build`) you get an error: 

```
transforming (916) node_modules/monaco-editor/esm/vs/base/browser/ui/progressbar/progressbar.css

rendering chunks (10)...error during build:
Error: Transform failed with 1 error:
assets/vendor.a83099bc.js:145514:5: error: Unexpected "."
```

If anyone has any ideas on how to get around this with Vite, I'd love to hear it...!

**Comment by suren-atoyan** [comment]
Hello @Ehesp 

Sorry for the late answer.
Could you please share a reproducible code snippet? I guess I know what can be the problem.

**Reply by suren-atoyan** [reply]
[The playground](https://github.com/suren-atoyan/monaco-react/tree/master/playground) is a `vite` app. You can check it, but just be sure that you import the Editor component from the `@monaco-editor/react`. So, replace [this](https://github.com/suren-atoyan/monaco-react/blob/master/playground/src/App.jsx#L3) line with `import Editor from '@monaco-editor/react'`. I've just checked it; it works as expected.

**Comment by Ehesp** [comment]
Yeah apologies it works fine. For some reason a totally unrelated issue caused the Vite Build to always stop on a file  within `monaco-editor`.
